tapir/coop/views.py

```diff
@@ -452,21 +459,29 @@ def get_queryset(self):
         if len(searches) == 1 and searches[0].isdigit():
             queryset = queryset.filter(pk=int(searches[0]))
         elif searches:
-            filter_ = Q(last_name__icontains="")
-            for search in searches:
-                search_filter = (
-                    Q(last_name__unaccent__icontains=search)
-                    | Q(first_name__unaccent__icontains=search)
-                    | Q(user__first_name__unaccent__icontains=search)
-                    | Q(user__last_name__unaccent__icontains=search)
-                )
-                filter_ = filter_ & search_filter
-
-            queryset = queryset.filter(filter_)
+            queryset = queryset.filter(
+                get_share_owner_name_filter(self.request.GET.get("search", ""))
+            )
 
         return queryset
 
 
+def get_share_owner_name_filter(search_string: str):
```

Comment:
Would it work to pull this in to a custom QuerySet? See `ShiftAttendanceQuerySet` (and `ShiftAttendance.objects`)